src/components/button-swap/button-swap.twig

```diff
@@ -0,0 +1,29 @@
+{#
+  This pattern currently only supports a start icon.
+  End icon support can be added in the future if needed.
+#}
+<section aria-labelledby="some-cool-id" class="c-button-swap js-button-swap{% if class %} {{ class }}{% endif %}">
+  <h3 class="u-hidden-visually" id="some-cool-id">THis is the name</h3>
```

Comment:
@cloudfour/dev For the `<section>` element to have a role of `region`, I need to give it an accessible name via an `aria` attribute _or_ explicitly set the `role` to `"region"`.

I'm curious what your thoughts are on which solution feels better. 

Original implementation:
- No heading existed
- The `<section>` element had an explicit `role="region"` attribute

New implementation:
- The `<section>` element has an `aria-labelledby` attribute but I'm unclear what the `aria-labelledby` value should point to.

Questions

1. Should the pattern include a visually hidden heading with a matching `id`?
    - Its seems the heading tag (e.g. `h3` or `h2`), `id` and heading text values should be properties that can be passed in.
2. If I add a visually hidden heading, how do I ensure the `id` is unique? I'm not sure. 🤔 
3. Should the pattern not have a visually hidden heading and instead require you pass in a value for the `aria-labelledby` putting the responsibility on the pattern consumer to create the element (with `id`) that this pattern should point to?
4. Should I avoid all of these challenges and explicitly set a `role="region"` value on the `<section>` element?  
    - While this _technically_ should work, it doesn't seem like the best UX when using an assistive technology like a screen reader. It'd be better if the region had an accessible name. So now that I think about it more, I feel this is not an acceptable solution.

Thoughts?

Reply:
Do the answers to any of these questions differ when we consider this within the context of the newsletter + notification sign-up UI? Since that is currently the only context for this component, just want to make sure we're considering that whole chunk of UI before we dive _too_ deeply into the questions of each individual pattern.

(If we're not sure, we may want to pause on developing individual components so we can prototype the whole experience of that element, _then_ carve off which piece or pieces deserve to be patterns.)

Reply:
I wonder if the `section` is actually adding helpful context here. If we can't think of a name that describes the section, then is it helpful to flag it as a region?

Going off of what Tyler said above, maybe this makes more sense when it is used as part of a larger form? In which case this component maybe doesn't need its own semantic role?

Reply:
> Going off of what Tyler said above, maybe this makes more sense when it is used as part of a larger form? In which case this component maybe doesn't need its own semantic role?

Fair! Thanks! 🙂